hosting/disk.go

```diff
@@ -49,3 +53,123 @@ type DiskImage struct {
 	Size     int
 	State    int
 }
+
+// CreateDisk creates a disk either from an existing one if `src` is not nil,
+// or an empty one if it is. `Name` is mandatory, `Size` defaults
+// to 10GB and `Region` to FR_SD3
+func (h Hostingv4) CreateDisk(disk DiskSpec, src *Disk) (Disk, error) {
+	var err error
+	if disk.RegionID == 0 {
+		disk.RegionID = defaultRegion
+	}
+	// Name is mandatory so we don't have to query the api again
+	// to get the name generated
+	if disk.Name == "" {
+		return Disk{}, errors.New("Disk name required")
+	}
+	if disk.Size == 0 {
+		disk.Size = defaultSize
+	}
+
+	response := []Operation{}
+	request := []interface{}{disk}
+	if src != nil {
+		request = append(request, []interface{}{src.ID})
+		err = h.Call("hosting.disk.create_from", request, &response)
+	} else {
+		err = h.Call("hosting.disk.create", request, &response)
+	}
+	if err != nil {
+		return Disk{}, err
+	}
+
+	d := Disk{
+		ID:       response[0].DiskID,
+		Name:     disk.Name,
+		Size:     disk.Size,
+		RegionID: disk.RegionID,
+		State:    "being_created",
+		Type:     "data",
+		BootDisk: false,
+	}
+	return d, nil
+}
+
+// DescribeDisks return a list of disks filtered with the options provided in `filter`
+// refer to the gandi api to see filtering options
+func (h Hostingv4) DescribeDisks(filter map[string]interface{}) ([]Disk, error) {
```

Comment:
We could also receive a DiskSpec struct and parse it into a map, not sure if it worth it